daffodil-runtime1/src/main/scala/org/apache/daffodil/infoset/SAXInfosetOutputter.scala

```diff
@@ -0,0 +1,213 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.daffodil.infoset
+
+import org.apache.daffodil.dpath.NodeInfo
+import org.apache.daffodil.xml.XMLUtils
+import org.xml.sax.helpers.AttributesImpl
+import org.xml.sax.ContentHandler
+import org.xml.sax.SAXException
+
+class SAXInfosetOutputter(ch: ContentHandler) extends InfosetOutputter {
+  /**
+   * Reset the internal state of this InfosetOutputter. This should be called
+   * inbetween calls to the parse method.
+   */
+  override def reset(): Unit = {
+    //do nothing
+  }
+
+  /**
+   * Called by Daffodil internals to signify the beginning of the infoset.
+   *
+   * @return true on sucess, false if there was an error and Daffodil should stop all
+   *         future calls to the InfosetOutputter
+   */
+  override def startDocument(): Boolean = {
+    try {
+      ch.startDocument
+      true
+    } catch {
+      case _: SAXException => false
+    }
+  }
+
+  /**
+   * Called by Daffodil internals to signify the end of the infoset.
+   *
+   * @return true on sucess, false if there was an error and Daffodil should stop all
+   *         future calls to the InfosetOutputter
+   */
+  override def endDocument(): Boolean = {
+    try {
+      ch.endDocument()
+      true
+    } catch {
+      case _: SAXException => false
+    }
+  }
+
+  /**
+   * Called by Daffodil internals to signify the beginning of a simple element.
+   *
+   * @param diSimple the simple element that is started. Various fields of
+   *                 DISimple can be accessed to determine things like the
+   *                 value, nil, name, namespace, etc.
+   * @return true on sucess, false if there was an error and Daffodil should stop all
+   *         future calls to the InfosetOutputter
+   */
+  override def startSimple(diSimple: DISimple): Boolean = {
+    val ns = if (diSimple.erd.thisElementsNamespace.isNoNamespace) "" else diSimple.erd.thisElementsNamespace.toString
+    val prefix = if (diSimple.erd.thisElementsNamespacePrefix == null) "" else diSimple.erd.thisElementsNamespacePrefix
+    val nsb = diSimple.namedQName
+    try {
+      if (prefix.nonEmpty && ns.nonEmpty) {
+        ch.startPrefixMapping(prefix, ns)
+      }
+
+      if (isNilled(diSimple)) { // TODO split this up into isNilled, hasValue or else (empty)
+        val attrs = new AttributesImpl()
+        attrs.addAttribute(XMLUtils.XSI_NAMESPACE, "nil", "xsi:nil", "", "true")
+        ch.startElement(ns, nsb.local, nsb.toQNameString, attrs)
+      } else if (diSimple.hasValue)  {
+        ch.startElement(ns, nsb.local, nsb.toQNameString, new AttributesImpl())
+        val text =
+          if (diSimple.erd.optPrimType.get.isInstanceOf[NodeInfo.String.Kind]) {
+            // TODO Should we use XMLInfosetOutputter remapped instead?
+            XMLUtils.remapXMLIllegalCharactersToPUA(diSimple.dataValueAsString)
```

Comment:
Yeah, I think mixing in XMLInfosetOutputter and using remapped seems reasonable.